api/internal/handler/global_rule/global_rule_test.go

```diff
@@ -0,0 +1,338 @@
+	mStore := &store.MockInterface{}
+	giveRet := `{
+		"id": "test",
+		"plugins": {
+			"jwt-auth": {}
+		}
+	}`
+	mStore.On("Get", mock.Anything).Run(func(args mock.Arguments) {
+		assert.Equal(t, "test", args.Get(0))
+	}).Return(giveRet, nil)
+
+	h := Handler{globalRuleStore: mStore}
+	r := gin.New()
+	h.ApplyRoute(r)
+
+	w := performRequest(r, "GET", "/apisix/admin/global_rules/test")
+	assert.Equal(t, 200, w.Code)
+}
+
+func TestHandler_Get(t *testing.T) {
+	tests := []struct {
+		caseDesc   string
+		giveInput  *GetInput
+		giveRet    interface{}
+		giveErr    error
+		wantErr    error
+		wantGetKey string
+		wantRet    interface{}
+	}{
+		{
+			caseDesc:   "normal",
+			giveInput:  &GetInput{ID: "test"},
+			wantGetKey: "test",
+			giveRet: `{
+				"id": "test",
+				"plugins": {
+					"jwt-auth": {}
+				}
+			}`,
+			wantRet: `{
+				"id": "test",
+				"plugins": {
+					"jwt-auth": {}
+				}
+			}`,
+		},
+		{
+			caseDesc:   "store get failed",
+			giveInput:  &GetInput{ID: "non-existent-key"},
+			wantGetKey: "non-existent-key",
+			giveErr:    fmt.Errorf("data not found"),
+			wantErr:    fmt.Errorf("data not found"),
+			wantRet: &data.SpecCodeResponse{
+				StatusCode: http.StatusNotFound,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.caseDesc, func(t *testing.T) {
+			getCalled := true
+			mStore := &store.MockInterface{}
+			mStore.On("Get", mock.Anything).Run(func(args mock.Arguments) {
+				getCalled = true
+				assert.Equal(t, tc.wantGetKey, args.Get(0))
+			}).Return(tc.giveRet, tc.giveErr)
+
+			h := Handler{globalRuleStore: mStore}
+			ctx := droplet.NewContext()
+			ctx.SetInput(tc.giveInput)
+			ret, err := h.Get(ctx)
+			assert.True(t, getCalled)
+			assert.Equal(t, tc.wantRet, ret)
+			assert.Equal(t, tc.wantErr, err)
+		})
+	}
+}
+
+func TestHandler_List(t *testing.T) {
+	tests := []struct {
+		caseDesc  string
+		giveInput *ListInput
+		giveData  []*entity.GlobalPlugins
+		giveErr   error
+		wantErr   error
+		wantInput store.ListInput
+		wantRet   interface{}
+	}{
+		{
+			caseDesc: "list all condition",
+			giveInput: &ListInput{
+				Pagination: store.Pagination{
+					PageSize:   10,
+					PageNumber: 1,
+				},
+			},
+			wantInput: store.ListInput{
+				PageSize:   10,
+				PageNumber: 1,
+			},
+			giveData: []*entity.GlobalPlugins{
+				{ID: "global-rules-1"},
+				{ID: "global-rules-2"},
+				{ID: "global-rules-3"},
+			},
+			wantRet: &store.ListOutput{
+				Rows: []interface{}{
+					&entity.GlobalPlugins{ID: "global-rules-1"},
+					&entity.GlobalPlugins{ID: "global-rules-2"},
+					&entity.GlobalPlugins{ID: "global-rules-3"},
+				},
+				TotalSize: 3,
+			},
+		},
+		{
+			caseDesc: "store list failed",
+			giveInput: &ListInput{
+				Pagination: store.Pagination{
+					PageSize:   10,
+					PageNumber: 10,
+				},
+			},
+			wantInput: store.ListInput{
+				PageSize:   10,
+				PageNumber: 10,
+			},
+			giveData: []*entity.GlobalPlugins{},
+			giveErr:  fmt.Errorf("list failed"),
```

Comment:
Why not return empty array?

Reply:
oh, it's a test case for error, no matter what is the data returned.